utils/extremum_tracking.hh

```diff
@@ -14,15 +14,16 @@
 namespace detail {
 
 template<typename T, typename Comparator>
+requires std::is_nothrow_copy_constructible_v<T> && std::is_nothrow_move_constructible_v<T>
```

Comment:
Update uses copy assignment operator, we should also require std::is_nothrow_copy_assignable_v<T>. In most cases the tracker probably operates on simple objects, but probably we could change that assignment to

`_value = std::move(value);`

In which case we'd need a std::nothrow_move_assignable_v<T> as well.

Reply:
Good idea

Reply:
I looked into this in v2, but the value can't always be moved.
I reduced the number of copies though to the minimum.